woocommerce/woocommerce
# Prevent Fatal Error when viewing order refund in admin page

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR prevent fatal error when viewing refund order with the `wp-admin/post.php?post=<REFUND_ID>&action=edit` URL and make the behaviour similar with the non-HPOS flow when viewing refund order.

This error occurred because the refund order is saved as `shop_order_placehold` type in the `wp_posts` when the sync is turned off. The type `shop_order_placehold` indicates that the redirect should occurred, however the refund order is not a valid type that have a UI page.

Closes #52802 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch
2. Create an Order, and refund the Order
3. Note down the refund ID
4. Open `wp-admin/post.php?post=<REFUND_ID>&action=edit`
5. Verify that the fatal error is not occurred, instead we got a message similar with what non-HPOS messaging 
<img width="837" alt="image" src="https://github.com/user-attachments/assets/7d42390b-7f2e-4b7e-b0a6-058bad178179">




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
